applications/StructuralMechanicsApplication/custom_constitutive/plasticity_isotropic_kinematic_j2.h

```diff
@@ -0,0 +1,399 @@
+#include "includes/constitutive_law.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class PlasticityIsotropicKinematicJ2
+ * @ingroup StructuralMechanicsApplication
+ * @brief Defines a Simo J2 plasticity CL with Isotropic & Kinematic Hardening in 3D
+ * @details This constitutive law is defined by the following parameters:
+ * - YOUNG_MODULUS
+ * - POISSON_RATIO
+ * - YIELD_STRESS
+ * - ISOTROPIC_HARDENING_MODULUS
+ * - REFERENCE_HARDENING_MODULUS (kinematic hardening modulus)
+ * - INFINITY_HARDENING_MODULUS (saturation hardening modulus)
+ * - HARDENING_EXPONENT
+ * @warning Only small strains (for the moment...)
+ * @author Fernando Rastellini
+ */
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) PlasticityIsotropicKinematicJ2
+    : public ConstitutiveLaw
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef ProcessInfo      ProcessInfoType;
+    typedef ConstitutiveLaw         BaseType;
+    typedef std::size_t             SizeType;
+
+    // Counted pointer of PlasticityIsotropicKinematicJ2
+    KRATOS_CLASS_POINTER_DEFINITION(PlasticityIsotropicKinematicJ2);
+
+    ///@}
+    ///@name Lyfe Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor.
+     */
+    PlasticityIsotropicKinematicJ2();
+
+    /**
+     * @brief Copy constructor.
+     */
+    PlasticityIsotropicKinematicJ2(const PlasticityIsotropicKinematicJ2& rOther);
+
+    /**
+     * @brief Destructor.
+     */
+    ~PlasticityIsotropicKinematicJ2() override;
+
+    /**
+     * @brief Clone function
+     * @return A pointer to a new instance of this constitutive law
+     */
+    ConstitutiveLaw::Pointer Clone() const override;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This function is designed to be called once to check compatibility with element
+     * @param rFeatures The Features of the law
+     */
+    void GetLawFeatures(Features& rFeatures) override;
+
+    /**
+     * @brief Dimension of the law:
+     */
+    SizeType WorkingSpaceDimension() override
+    {
+        return 3;
+    };
+
+    /**
+     * @brief Voigt tensor size:
+     */
+    SizeType GetStrainSize() override
+    {
+        return 6;
+    };
+
+    /**
+     * @brief Returns whether this constitutive Law has specified variable (bool)
+     * @param rThisVariable the variable to be checked for
+     * @return true if the variable is defined in the constitutive law
+     */
+    bool Has(const Variable<bool>& rThisVariable) override;
+
+    /**
+     * @brief Returns the value of a specified variable (bool)
+     * @param rThisVariable the variable to be returned
+     * @param rValue a reference to the returned value
+     * @return rValue output: the value of the specified variable
+     */
+    bool& GetValue(
```

Comment:
are sure sure this needs to be by-reference?

Reply:
actually i think it is wrong

Reply:
it was wrong.
corrected...